File: by-python/py-boj/3273-두 수의 합.py
n = int(input())
numbers = list(map(int, input().split()))
target = int(input())

numbers.sort()

lo, hi = 0, len(numbers) - 1

answer = 0
while lo < hi:
    sum_two = numbers[lo] + numbers[hi]
    if sum_two == target:
        answer += 1
        lo += 1

    elif sum_two < target:
        lo += 1
    elif sum_two > target:
        hi -= 1

print(answer)

File: by-kotlin/kt-boj/solutions/3273-두 수의 합.kt
import java.io.BufferedReader
import java.io.InputStreamReader

private var n = 0
private var x = 0
private lateinit var numbers: Array<Int>


fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    n = readLine().toInt()
    numbers = readLine().split(" ").map { it.toInt() }.toTypedArray()
    x = readLine().toInt()

    numbers.sort()

    var lo = 0
    var hi = numbers.size - 1

    var answer = 0
    while (lo < hi) {
        var sumOfTwo = numbers[lo] + numbers[hi]
        when {
            sumOfTwo == x -> {
                answer++
                lo++
            }
            sumOfTwo < x -> {
                lo++
            }
            else -> {
                hi--
            }
        }
    }
    print(answer)
}
